inspirehep/modules/records/serializers/fields_export.py

```diff
@@ -0,0 +1,318 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014-2017 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Instructions on where to find relevant bits of data in API."""
+
+from __future__ import absolute_import, division, print_function
+from idutils import is_arxiv_post_2007
+from inspire_utils.record import get_value
+from inspire_utils.date import PartialDate
+from inspirehep.modules.hal.utils import get_page_artid, get_conference_record, get_conference_title
+from babel import Locale
+from isbn_hyphenate import hyphenate, IsbnError
+
+
+MAX_AUTHORS_BEFORE_ET_AL = 10  # According to CSE stylebook
+
+# Fields shared among all bibtex entries:
+COMMON_FIELDS_FOR_ENTRIES = ['key', 'SLACcitation', 'archivePrefix', 'doi', 'eprint', 'month', 'note', 'primaryClass',
+                             'title', 'url', 'year']
+
+# Fields for a given bibtex entry. Since we're trying to match as many as possible
+# it doesn't matter whether they're mandatory or optional
+FIELDS_FOR_ENTRY_TYPE = {
+    'techreport': ['author', 'collaboration', 'number', 'address', 'type', 'institution'],
+    'phdthesis': ['reportNumber', 'school', 'address', 'type', 'author'],
+    'inproceedings': ['publisher', 'author', 'series', 'booktitle', 'number', 'volume', 'reportNumber', 'editor',
+                      'address', 'organization', 'pages'],
+    'misc': ['howpublished', 'reportNumber', 'author'],
+    'mastersthesis': ['reportNumber', 'school', 'address', 'type', 'author'],
+    'proceedings': ['publisher', 'series', 'number', 'volume', 'reportNumber', 'editor', 'address', 'organization',
+                    'pages'],
+    'book': ['publisher', 'isbn', 'author', 'series', 'number', 'volume', 'edition', 'editor', 'reportNumber',
+             'address'],
+    'inbook': ['chapter', 'publisher', 'author', 'series', 'number', 'volume', 'edition', 'editor', 'reportNumber',
+               'address', 'type', 'pages'],
+    'article': ['author', 'journal', 'collaboration', 'number', 'volume', 'reportNumber', 'pages'],
+}
```

Comment:
Mh, these global variables make me think that we should structure a little better the `records` module, so that this configuration lives in a single place. See: https://github.com/inspirehep/inspire-next/blob/67ff20df8c6ed9abb6a7d59a6a7c536d2610a925/inspirehep/modules/hal/config.py

Reply:
This lack of structure also led to defining `MAX_AUTHORS_BEFORE_ET_AL` in two places!